src/main/java/org/breedinginsight/services/TraitUploadService.java

```diff
@@ -90,49 +95,46 @@ public ProgramUpload updateTraitUpload(UUID programId, CompletedFileUpload file,
                 traits = parser.parseCsv(file.getInputStream());
             } catch(IOException | ParsingException e) {
                 log.error(e.getMessage());
-                throw new UnprocessableEntityException("Error parsing csv: " + e.getMessage());
+                throw new HttpBadRequestException("Error parsing csv: " + e.getMessage());
             }
         } else if (mediaType.getName().equals(MediaType.XLS) ||
                    mediaType.getName().equals(MediaType.XLSX)) {
             try {
                 traits = parser.parseExcel(file.getInputStream());
             } catch(IOException | ParsingException e) {
                 log.error(e.getMessage());
-                throw new UnprocessableEntityException("Error parsing excel: " + e.getMessage());
+                throw new HttpBadRequestException("Error parsing excel: " + e.getMessage());
+            } catch (ValidatorException e){
```

Comment:
Since this is being rethrown and not allowed to flow past this any ValidatorException from the TraitFileParser will short circuit and not process other trait errors. For example if you have an invalid scale type and then other traits with say missing required fields, you'll only see the scale type error and not the others. Didn't we still want to check those other traits as well? I attached a file for an example.

[data_one_row_invalid_scale.xlsx](https://github.com/Breeding-Insight/bi-api/files/5007999/data_one_row_invalid_scale.xlsx)

Reply:
That was intentional because the type parsing errors may lead to false errors in the missing field validation check. For example, if a scale class was passed, but is a bad type, it will leave the scale class blank. If we let that keep going in the validation, it will throw a missing scale error, which would be wrong. 

So if the controller can't parse the file because of bad data types it will throw an error before moving onto other checks.

Reply:
That makes sense for the particular trait that had the invalid scale, but couldn't you just drop that trait from further checks and check the other traits?

Reply:
You could, but that trait may have errors such as missing trait name that wouldn't be included in the initial error return because it was ignored. It would then show up as an error when the user submitted the second time which may be confusing to the user because it didn't appear initially among the errors of similar type. 

Looking for @Breeding-Insight/dev-team feedback on this. Nick and I discussed this and didn't come to a conclusion. One thing we agreed on was that the ideal solution would require one large validation method (combination of all of the validations) that would require a lot of work and may not be worth it right now.

Reply:
Decided in team meeting to keep it how it is and implemented the ideal solution when we work on the trait editing page for trait imports.